Removing cart warnings/errors display

juanpprieto opened the pull request:
### WHY are these changes introduced?

This PR addresses TypeScript compilation errors that occurred after removing CartUserErrors and CartWarnings components from the skeleton template, and includes related cookbook maintenance updates.

### WHAT is this pull request doing?

#### Core Fix: TypeScript Error Resolution
- Stop propagating `warnings` and `userErrors` to CartMain.
- Removed up imports and references to deleted `CartUserErrors.tsx` and `CartWarnings.tsx`
- Ensured skeleton template compiles without TypeScript and build errors

#### Cookbook Recipe Updates  

Updated affected recipes

#### Changeset Cleanup
- **Removed inaccurate changeset**: Deleted `cart-user-errors-display.md` that incorrectly claimed to add components that were actually removed

#### Validation Checklist
- [ ] Template compiles without TypeScript errors
- [x] Template builds successfully
- [x] Development server starts without issues
- [x] No references to removed CartUserErrors/CartWarnings components

#### Checklist

- [x] I've read the [Contributing Guidelines](https://github.com/Shopify/hydrogen/blob/main/CONTRIBUTING.md)
- [x] I've added or updated the documentation

juanpprieto committed:
Fix TypeScript error in CartSummary after removing warnings/errors

The refactoring that removed cart warnings and errors inadvertently
changed CartSummary to use CartMainProps, which expects
CartApiQueryFragment | null. However, CartSummary receives the result
of useOptimisticCart(), which returns OptimisticCart<CartApiQueryFragment | null>.

This commit restores the correct CartSummaryProps type to accept
OptimisticCart, resolving the TypeScript compilation error.

juanpprieto committed:
fix(cookbook): Update express recipe for skeleton compatibility

- Removed non-existent files (CartUserErrors.tsx, CartWarnings.tsx) from deletedFiles list
- Updated all patch files to work with current skeleton template
- Fixed line numbers and context to match skeleton structure
- Successfully regenerated recipe with new patch hashes
- Recipe applies cleanly to skeleton template without creating .orig files

juanpprieto committed:
Revert "fix(cookbook): Update express recipe for skeleton compatibility"

This reverts commit fe00a98cb5a5b0ae711570a378140aff5fb5d917.

juanpprieto committed:
commit remaining changes after revert

juanpprieto committed:
fix(cookbook): Update markets recipe for skeleton compatibility

- Fixed CartMain.tsx patch to match current skeleton import structure
- Updated all patch files to work with current skeleton template
- Fixed line numbers and context to match skeleton structure
- Successfully regenerated recipe with new patch hashes
- Recipe applies cleanly to skeleton template without creating .orig files

juanpprieto committed:
Revert "fix(cookbook): Update markets recipe for skeleton compatibility"

This reverts commit b7b59a7ef7f2cdb1d7d27234820488dd89fabc63.

juanpprieto committed:
commit all cookbook changes after revert

juanpprieto committed:
fix(cookbook): Update multipass recipe for skeleton compatibility

- Fixed CartSummary.tsx patch to match current skeleton import structure
- Removed references to non-existent CartWarnings and CartUserErrors components
- Updated all patch files to work with current skeleton template
- Fixed line numbers and context to match skeleton structure
- Successfully regenerated recipe with new patch hashes
- Recipe applies cleanly to skeleton template without creating .orig files
- Updated LLM prompt files and cookbook metadata

juanpprieto committed:
Remove inaccurate changeset for CartUserErrors feature

The cart-user-errors-display.md changeset claimed to add CartUserErrors
components that were actually removed from the skeleton template.
Deleting this changeset to prevent inaccurate release notes.

juanpprieto committed:
fix(cookbook): Remove non-existent CartUserErrors.tsx from express recipe deletedFiles

- Removed CartUserErrors.tsx and CartWarnings.tsx from express recipe deletedFiles list
- These components no longer exist in the skeleton template after removal in ff1edd0d4
- Fixes CI validation error: ENOENT: no such file or directory, unlink CartUserErrors.tsx
- Express recipe now validates successfully without trying to delete non-existent files
- Updated package-lock.json with latest dependencies

juanpprieto committed:
fix(cookbook): Fix markets recipe CartMain.tsx patch compatibility

- Updated CartMain.tsx patch to match current skeleton import structure
- Removed reference to non-existent CartWarning types that were removed in ff1edd0d4
- Fixed patch context to match actual skeleton structure without CartWarnings imports
- Markets recipe now validates successfully without creating .orig files
- Resolves CI validation error with fuzz 2 application

kdaviduik commented on cookbook/llms/express.prompt.md:
```diff
@@ -77,7 +77,7 @@ Key changes:
 
 > [!NOTE]
-> Compatible with React Router 7.9.x
+> Compatible with React Router 7.8.x
```
```suggestion
> Compatible with React Router 7.9.x
```

kdaviduik commented on templates/skeleton/app/components/CartUserErrors.tsx:
```diff
@@ -1,38 +0,0 @@
-  }
-
-  return (
-    <div className="cart-user-errors" role="alert">
```
we need to update the CSS file to remove the styles for this and the warnings

juanpprieto committed:
Update React Router version references from 7.8.x to 7.9.x

Updates version references throughout the codebase to reflect the current
React Router version being used:

Cookbook files:
- recipe.yaml files for express, combined-listings, and multipass recipes
- LLM prompt files for express and multipass
- README patch for multipass recipe

Source code:
- CLI version check messaging
- TypeScript comments and documentation
- Route generation comments

Also removes outdated migrate-example.md file.

juanpprieto committed:
Remove userErrors and Warnings styles

juanpprieto committed:
fix(cookbook): Update express recipe for skeleton compatibility

- Manually resolved CSS patch conflict in app.css
- Updated patch from wholesale file replacement to proper incremental change
- Successfully regenerated all recipe patches with new hashes
- Recipe now applies cleanly without creating .orig/.rej files
- All tests pass: install, codegen, typecheck, build, validation
- Updated LLM prompt files and cookbook metadata

juanpprieto committed:
Update lock

juanpprieto committed:
fix(cookbook): Remove incorrectly committed guide files

- Remove enhanced-recipe-fix-guide.md and recipe-validation-plan.md
- These should be in .claude/commands/ not in cookbook directory
- Express recipe fix remains intact and functional

kdaviduik reviewed (approved):
LGTM!
